spring-integration-redis/src/main/java/org/springframework/integration/redis/util/RedisLockRegistry.java

```diff
@@ -260,9 +258,34 @@ public void destroy() {
 		}
 	}
 
-	private final class RedisLock implements Lock {
+	public enum RedisLockType {
+		PUB_SUB_LOCK, SPIN_LOCK;
+	}
+
+	private Function<String, RedisLock> getRedisLockConstructor(@NonNull RedisLockType redisLockType) {
+		return switch (redisLockType) {
+			case PUB_SUB_LOCK -> RedisPubSubLock::new;
+			case SPIN_LOCK -> RedisSpinLock::new;
```

Comment:
We cannot use switch expression because we have to back-port to `5.5.x` which is still based on Java 8.